src/csv.ts

```diff
@@ -164,4 +164,28 @@ export const parseCSVContent = <T extends z.ZodType>(csvContent: string, schema:
 export const parseCSV = async <T extends z.ZodType>(csv: File, schema: T): Promise<Result<T>> => {
     const csvContent = await getCSVContent(csv);
     return parseCSVContent(csvContent, schema)
+}
+
+type ResultSingleRow<T extends z.ZodType> = {
+    success: true,
+    row: z.infer<T>,
+} | {
+    success: false,
+    error: z.ZodError<T>[]
+}
+
+export const parseCSVRow = <T extends z.ZodType>(row: string, schema: T): ResultSingleRow<T> => {
+    const enrichedRow = `\n${row}`
+    const errors = getCSVErrors(enrichedRow, schema);
+    if (errors?.rows) {
+        return {
+            success: false,
+            error: Object.values(errors.rows)
+        }
+    }
+
+    return {
+        success: true,
+        row: getRowsFromCSV(enrichedRow, schema)[0]
```

Comment:
@bartoszgolebiowski the only thing that really stands out to me here is that it looks like the data is effectively parsed twice, assuming no errors were found in the first pass, and it looks like this is the same for `parseCSVContent`. So one improvement might be to re-architect things a bit -- i.e. it can be assumed that if no errors were detected then the results are correct so they can be returned as successful without the need to parse again?

Reply:
You are right. Instead of parsing twice, I will collect errors and valid rows in the same function! 
I also updated some docs.